### PR TITLE
Add fixes for local dev setup, and reorg file stucture

### PR DESCRIPTION
**Pull request by lilbitner**
## Purpose

This PR adjusts a few configurations with the `lambda` of the Slack app, to help support local/future development. 

## Approach

The following changes were made: 

1. All docker related files were moved within the `lambda` directory. 
2. The docker commands and logic for 1. standing up `DynamoDB` 2. automatically running the table migration as well as 3. starting the local instance of `DynamoDB` admin, have been added as part of the docker workflow. A lot of this logic is copied from the GA4 setup (s/o @ghepting  + @matthew-gordon work there). 
3. Some adjustments have been added to the `README` to help facilitate other developers running this app locally with efficiency (some follow-up work to the great adjustments @mgoudy91 made in an previously unmerged PR). (I have added more Integrations team specific information to the internal Wiki page around setting this up with our accounts/apps etc). 

## Testing steps

Feel free to pull this code down and test locally that you are able to standup everything correctly! Would appreciate any feedback on this process. 

## Breaking Changes

This is where my dev-ops knowledge is somewhat limited. While I did not make direct changes to production `yml` files, I am assuming changes to the `Make` and `Dockerfile` affect production workflows. Everything did build in this PR appropriately but am curious if there is a good way to test this? Any recommendations?

### CODE REVIEW

**Review by ryunsong-contentful — APPROVED** [review]
Nice

File: apps/slack/README.md [hunk 1]
```diff
@@ -101,17 +102,16 @@ This sections explains how to run the Slack app locally.
...
-- Start a docker container with a local DynamoDB instance using `make start-backend`
-- If you start the project for the first time, run `yarn create_table` in the container
-- Run `yarn start:lambda` in the container
+- Start a docker container with a local DynamoDB instance using `make go` from within the lamda directory
 
 ### Frontend
 
-- Set the following variables in `frontend/.env.development`:
-  - `REACT_APP_SLACK_CLIENT_ID`: The Slack app Client ID (Basic Information -> App Credentials -> Client ID)
+- Set the following variables in `frontend/.env.development` using the `.env.development.example` file:
```
Comment (mgoudy91) [inline apps/slack/README.md:114]: 👏

**Review by mgoudy91 — APPROVED** [review]
I think this looks really great. Awesome work synthesizing all these disparate and incomplete understandings of the issues with the app into a holistic solution!

**Comment by mgoudy91** [comment]
I think your question about prod testing is a good one (and not dissimilar from the serverless 2 -> 3 upgrade) Ideally we would have a fleshed out staging env that would be perfect for this. As is, we'll likely need to have a manually observed deployment with a lot of emphasis on being ready to roll back. Worth a discussion with the team I think.

**Pull request merged**